emission/tests/modellingTests/TestSimilarityMetric.py

```diff
@@ -6,26 +6,93 @@ class TestSimilarityMetric(unittest.TestCase):
-        similar = metric.similar(coords0, coords1, similarity_threshold)
-        self.assertTrue(similar)
+        coords1 = metric.extract_features(trips[1])        
+        similarOD1 = metric.similar(coords0, coords1, similarity_threshold)
+        similarOD2 = metric.similar(coords0[:2], coords1[:2], similarity_threshold)
+        similarOD3 = metric.similar(coords0[2:], coords1[2:], similarity_threshold)
+
+        ## Sub-Test 4 :
+        # random, but, only origin points are sampled within a circle and should always be < sim threshold
+        # Since origin of two points lies within threshold limits,they should be similar
+        # when we check just origin for similarity.
+
+
+        trips = etmm.generate_mock_trips('alice', 2, [0, 0], [1, 1], 'o_',threshold=generate_points_thresh)        
+        coords0 = metric.extract_features(trips[0])[:2]
+        coords1 = metric.extract_features(trips[1])[:2]        
+        similarO = metric.similar(coords0, coords1, similarity_threshold)
+
+        ##Sub-Test 5 :
+        # random, but, only destination points are sampled within a circle and should always be < sim threshold
+        # Since destination of two points lies within threshold limits,they should be similar
+        # when we check just destination for similarity.
+
+        trips = etmm.generate_mock_trips('Caty', 2, [0, 0], [1, 1], '_d',threshold=generate_points_thresh)        
+        coords0 = metric.extract_features(trips[0])[2:]
+        coords1 = metric.extract_features(trips[1])[2:]        
+        similarD = metric.similar(coords0, coords1, similarity_threshold)
+
+        # All the similars must be true
+        self.assertTrue(similarOD1) # RESULT SUB-TEST 1
+        self.assertTrue(similarOD2) # RESULT SUB-TEST 2
+        self.assertTrue(similarOD3) # RESULT SUB-TEST 3
+        self.assertTrue(similarO)  # RESULT SUB-TEST 4
+        self.assertTrue(similarD) # RESULT SUB-TEST 5
     
     def testODsAreNotSimilar(self):
         generate_points_thresh = 0.001  # approx. 111 meters
-        similarity_threshold = 500  # 
-        
-        trips0 = etmm.generate_mock_trips('bob', 1, [0, 0], [1, 1], threshold=generate_points_thresh)
-        trips1 = etmm.generate_mock_trips('alice', 1, [2, 2], [3, 3], threshold=generate_points_thresh)
+        similarity_threshold = 111  # 
+        metric = eamso.OriginDestinationSimilarity()
+
+        ## Sub-Test 1-2: 
+        # Two trips with neither origin nor destination coordinates within threshold
+        # must not be similar in any configuration of similarity testing.
+        trips = etmm.generate_mock_trips('bob', 2, [0, 0], [1, 1], '__', threshold=generate_points_thresh)  
+        coords0 = metric.extract_features(trips[0])
+        coords1 = metric.extract_features(trips[1])
+        similar11 = metric.similar(coords0[:2], coords1[:2], similarity_threshold)
+        similar12 = metric.similar(coords0[2:], coords1[:], similarity_threshold)
```

Comment:
shouldn't this be `coords[:2]`?

Reply:
I guess theres no variable by the name `coords`. However, it's still wrong there. It should be

```similar12 = metric.similar(coords0[2:], coords1[2:], similarity_threshold)``` . the 2 is missing in the second coords1 .

Reply:
correct. I don't claim that my reviews are ready to copy and paste 😄

Reply:
Yeah. I did. I was working on its last bit during that time.